app/src/main/java/org/breezyweather/common/ui/activities/AlertActivity.kt

```diff
@@ -133,7 +133,18 @@ class AlertActivity : GeoActivity() {
                 finish()
                 return@LaunchedEffect
             }
-            location.value = locationC
+
+            // Astro data is needed to check if the sun is still up or if it has set when day/night
+            // mode per location is enabled.
+            location.value = locationC.copy(
+                weather = weatherRepository.getWeatherByLocationId(
+                    locationC.formattedId,
+                    withDaily = true,
+                    withHourly = false,
+                    withMinutely = false,
+                    withAlerts = false
```

Comment:
Maybe we should take this opportunity to refactor and use `withAlerts = true` and remove the second query of alerts later.

Reply:
Thanks for the hint. I updated the PR accordingly.